generate_adapter.py

```diff
@@ -73,11 +73,13 @@ def main(
         print("Loading model ...", file=sys.stderr)
         t0 = time.time()
         model = LLaMA(LLaMAConfig())  # TODO: Support different model sizes
+
         # 1. Load the pretrained weights
         pretrained_checkpoint = torch.load(pretrained_path)
         model.load_state_dict(pretrained_checkpoint, strict=False)
         # 2. Load the fine-tuned adapter weights
-        adapter_checkpoint = torch.load(adapter_path)
+        adapter_checkpoint = torch.load(adapter_path, map_location=torch.device("cpu"))
```

Comment:
doing this because the checkpoint was saved in bfloat16 with device=cuda. If we don't map the location to cpu, we would load the checkpoint in to GPU memory. If we didn't apply quantization, this would be fine. But if we apply quantization in the model, the float16 weights must be converted anyway. So we don't want to spike the memory unnecessarily.